### PR TITLE
[ENGNP-442] Migrate from Travis to Github Actions

### PR DESCRIPTION
**Pull request by ChristianSassine**
Adds Github Actions workflow to replace travis' CI.

Refs : [ENGNP-442]

[ENGNP-442]: https://adgear.atlassian.net/browse/ENGNP-442?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        with:
+          path: |
+           ~/.cache/rebar3
+           _build
+          key: ci-${{runner.os}}-${{env.cache-name}}-erlang_${{matrix.erlang}}-rebar3-${{hashFiles('rebar.lock')}}
+          restore-keys: |
+            ci-${{runner.os}}-${{env.cache-name}}-erlang_${{matrix.erlang}}-rebar3
+            ci-${{runner.os}}-${{env.cache-name}}-erlang_${{matrix.erlang}}
+      - name: elvis
+        run: make elvis
```
Comment (ChristianSassine) [inline .github/workflows/ci.yml:25]: Separated `make test` into individual `make` steps to have an easier way to identify which step failed if a failure occurs
image

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+name: Test CI
+on: push
+jobs:
+  Test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        erlang: ['21.1', '20.3']
```
Comment (berbiche) [inline .github/workflows/ci.yml]: Could you update the test Matrix to 
```suggestion
        erlang: ['24', '25', '26']
```

The lowest version we use in our other projects is OTP 24 and will eventually upgrade to 25 or 26.

**Review by berbiche — CHANGES_REQUESTED** [review]
Nice work!
Thanks for updating the test setup 😄

File: .github/workflows/ci.yml (hunk 2 above)
Comment (ChristianSassine) [reply .github/workflows/ci.yml]: Yup. I'm trying to use the newest versions locally but it seems to be only working for OTP 24. OTP 25 and OTP 26 give me the same error as in the [GHA](https://github.com/adgear/consent-string/actions/runs/5282129386/jobs/9556648670) : 
image

Comment (pdeschen) [reply .github/workflows/ci.yml]: Focus. Ditch OTP 26 out if it's not working as-is right now. Let the team do the eventual OTP 26 migration. Here, this is about travis migration.

Comment (ChristianSassine) [reply .github/workflows/ci.yml]: Alright, I'll keep it at OTP 24 since it's the lowest version that works and that we use.

**Review by berbiche — APPROVED** [review]
LGTM!
Thanks

**Pull request merged**